train_cnn.py

```diff
@@ -0,0 +1,118 @@
+                layers.Dense(7 * 7 * 16, 10, activation=act.relu)
+        )
+        """
+
+        self.seq_layers = self.sequential(
+                layers.Conv2D(3, 32, kernel_size=3, strides=1, padding='valid', activation=act.relu),
+                layers.MaxPool2D(pool_size=3, strides=2, padding='valid'),
+                layers.Conv2D(32, 64, kernel_size=3, strides=1, padding='valid', activation=act.relu),
+                layers.MaxPool2D(pool_size=3, strides=2, padding='valid'),
+                layers.Conv2D(64, 64, kernel_size=3, strides=1, padding='valid', activation=act.relu),
+                layers.MaxPool2D(pool_size=3, strides=2, padding='valid'),
+                layers.Conv2D(64, 128, kernel_size=3, strides=1, padding='valid', activation=act.relu),
+                layers.MaxPool2D(pool_size=3, strides=2, padding='valid'),
+                layers.Conv2D(128, 128, kernel_size=3, strides=1, padding='valid', activation=act.relu),
+                layers.MaxPool2D(pool_size=3, strides=2, padding='valid'),
+                layers.AvgPool2D((1,1))
+        )
+
+        self.dnnModel = self.sequential(
+                torch.nn.Linear(128, 64),
+                torch.nn.Linear(64, 32),
+                torch.nn.Linear(32, 8)
+        )
+
+        def forward(
+                self,
+                x
+        ):
+            output = self.seq_layers(x)
+            output = output.squeeze()
+            output = self.dnnModel(output)
+            return output
+
+        """
+        def forward(
+                self,
+                x
+        ):
+            o = self.seq_layers.forward(x)
+            return o
+        """
+
+"""
+# Load dataset
+train = pd.read_csv('/home/khushi/Documents/simple-neural-network/datasets/data/mnist_train.csv').astype(np.float32)
+train_data = pd.DataFrame(train.iloc[:, 1:])
+train_target = pd.DataFrame(train.iloc[:, 0])
+
+# Converting into tensors.
+train_data = torch.tensor(train_data.values)
+train_target = torch.tensor(train_target.values)
+
+# Normalizing the input pixel/image.
+train_data = torch.nn.functional.normalize(train_data, p=2.0)
+train_target = torch.nn.functional.normalize(train_target, p=2.0)
+
+train_loader = dataloader.DataLoader(train_data, train_target, batch_size=64)
+"""
+
+transform = transforms.Compose([
+    transforms.Resize((224, 224)),
+    transforms.ToTensor(),
+    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+])
+
+dataset = datasets.ImageFolder(root = '/home/khushi/Documents/simple-neural-network/datasets/data/archive/natural_images', transform = transform)
+dataloader = torch.utils.data.DataLoader(dataset, batch_size = 32) #, shuffle = True)
+
+cnn = CNN()
+opt = optimizers.Adam(cnn.params, learning_rate=0.001)
+loss = losses.SparseSoftMaxCrossEntropyWithLogits()
+
+for step in range(100):
+    bx, by = dataloader.next_batch()
+    by_ = cnn.forward(bx)
+    _loss = loss(by_, by)
+    cnn.backward(_loss)
```

Comment:
cnn would keep the track of params. no need to pass _loss
You should rather do
```suggestion
    _loss.backward()
```